pkg/dispatchcli/cmd/update_baseimage.go

```diff
@@ -0,0 +1,147 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vmware/dispatch/pkg/dispatchcli/i18n"
+	"github.com/vmware/dispatch/pkg/image-manager/gen/client/base_image"
+	"github.com/vmware/dispatch/pkg/image-manager/gen/models"
+)
+
+var (
+	updateBaseImageLong    = "Updates a base image from a given json representation"
+	updateBaseImageExample = `{
+		"dockerUrl": "vmware/dispatch-openfaas-nodejs6-base:0.0.3-dev1",
+		"groups": null,
+		"language": "nodejs6",
+		"name": "nodejs6-base",
+		"public": true,
+		"reason": null,
+		"tags": [
+			{
+				"key": "role",
+				"value": "test"
+			}
+		]
+	}`
+	updateFilePath          string
+	imageURLUpdateValue     []string
```

Comment:
This is much more pertinent to a boolean valued flag, but the issue here comes down to the semantics of the update command and its flags. For instance:
`dispatch update base-image base-nodejs6 --language "SOME_LANGUAGE"`
I would assume this command would update only the language property of the base-nodejs6 base image and leave all other properties unchanged.

When a flag is not specified Cobra populates the associated variable with some default value. A reasonable default might be the empty string `""`. So in the above case the default value for the imageURLUpdateValue would be come `""`. But it is possible that a user might want to do something like `dispatch update base-image base-nodejs6 --image-url ""` in which case they want to set the image-url to the empty string. So I need some way of determining whether the flag wasn't used, or if it was used to change to a value to what we think is the default.

The above case is not the best example. It probably doesn't make sense to let a user input a blank image url. I did this because the create command does not error on an empty value for the image url.

The real issue is with commands that are boolean variables. If we want the semantics to be that you use the flags that correspond to fields you want to update, then when someone uses (for instance) --public that means that we should update the public field. But if this is just a boolean, how do we let them specify that they want it to be false? That's why I made publicFlagUpdateValue a string array as well. Now I can set the default to an empty array when the flag is not used, which means no update. A user can then specify `--public true` or `--public false` to actually make the change.

I realize that public is being removed from this example, but in the case of boolean flags for other resources I'm presenting it here to get feedback on how this should work.

Reply:
Ok, I see your logic.  How about having a "DEFAULT_VALUE" or some other constant as the default value to address this.  Otherwise, you've change the behavior of the CLI and now, the user may think that theses flags can be issued multiple times.  However, if it's invisible to the user, maybe what you're proposing is correct.  Perhaps reach out to @kars7e or someone else on the team.  I'm not entirely sure what the best approach is.  I would probably go the "DEFAULT_VALUE" route myself.

Reply:
Looks like you're hitting this: https://github.com/spf13/cobra/issues/23. The solution mentioned there is to use [`Changed()` method](https://godoc.org/github.com/spf13/pflag#FlagSet.Changed) which returns true if the flag was provided in CLI.